sky/backends/backend_utils.py

```diff
@@ -2373,11 +2373,20 @@ def check_cluster_available(
     bright = colorama.Style.BRIGHT
     reset = colorama.Style.RESET_ALL
     if handle is None:
-        if previous_cluster_status is None:
-            error_msg = f'Cluster {cluster_name!r} does not exist.'
-        else:
-            error_msg = (f'Cluster {cluster_name!r} was preempted or manually '
-                         'terminated in console.')
+        error_msg = (f'Cluster {cluster_name!r} not found on the cloud '
+                     'provider.')
+        if previous_cluster_status is not None:
+            assert record is not None, previous_cluster_status
+            actions = []
+            if record['handle'].launched_resources.use_spot:
+                actions.append('preempted')
+            if record['autostop'] > 0 and record['to_down']:
+                actions.append('autodowned')
+            actions.append('manually terminated in console')
+            if len(actions) > 1:
+                actions[-1] = 'or ' + actions[-1]
+            actions_str = ', '.join(actions)
+            error_msg += (f' It was either {actions_str}.')
```

Comment:
```suggestion
            error_msg += (f' It was likely {actions_str}.')
```
to handle 1 action.

Reply:
Hmm, when 1 action only "likely manually terminated" sounds good since it is indeed a guess.  When >1 actions, e.g., "likely autodowned or manually terminated" seems less certain than we really mean. This is extremely nitpicky, but perhaps when >1 actions we can use "either". Up to you.